internal/clientconn/conninfo/conn_info.go

```diff
@@ -34,6 +39,25 @@ type ConnInfo struct {
 	rw       sync.RWMutex
 	username string
 	password string
+
+	curRW         sync.RWMutex
+	cursors       map[int64]Cursor
+	cursorCounter int64
+}
+
+// NewConnInfo return a new ConnInfo.
+func NewConnInfo() *ConnInfo {
+	return &ConnInfo{
+		cursors:       map[int64]Cursor{},
+		cursorCounter: 1,
+	}
+}
+
+// Cursor represents a cursor.
+type Cursor struct {
+	tx     pgx.Tx
```

Comment:
With #1947, we could store only one iterator with a custom close function that would commit tx